src/StdLib.jl

```diff
@@ -112,10 +136,35 @@ Base.cmp(a::String, b::CppBasicString) = cmp(a,String(b))
 
 # Make sure functions taking a C++ string as argument can also take a Julia string
 CxxWrapCore.map_julia_arg_type(x::Type{<:StdString}) = AbstractString
-StdString(x::String) = StdString(x,ncodeunits(x))
-StdLib.StdStringAllocated(x::String) = StdString(x,ncodeunits(x))
-Base.cconvert(::Type{CxxWrapCore.ConstCxxRef{StdString}}, x::String) = StdString(x,ncodeunits(x))
-Base.cconvert(::Type{StdLib.StdStringDereferenced}, x::String) = StdString(x,ncodeunits(x))
+
+"""
+    StdString(str::String)
+
+Create a new `StdString` from the contents of the string. Any null-characters will be
+included in the string such that `ncodeunits(str) == ncodeunits(StdString(str))`.
+"""
+StdString(x::String) = StdString(x, ncodeunits(x))
+
+"""
+    StdString(str::Union{Cstring, Base.CodeUnits, Vector{UInt8}, Ref{Int8}, Array{Int8}})
+
+Create a new `StdString` from the null-terminated character sequence.
+
+## Examples
+
+```julia
+julia> StdString(b"visible\0hidden")
+"visible"
+```
+"""
+StdString(::Union{Cstring, Base.CodeUnits, Vector{UInt8}, Ref{Int8}, Array{Int8}})
```

Comment:
this was initially surprising to me but I think it's consistent with the `std::string` constructor behavior: if you pass in a vector of bytes (without lenght), it will treat it as null-terminated c string.  might be worth calling out how to construct the non-null-terminated form here as well (i.e., use a `String`, or pass in the lenght as teh second argument)